### PR TITLE
Collectives Ops : Match example from the StableHLO op description comment with example from the spec

### PR DESCRIPTION
**Pull request by abhigunj**
as per https://github.com/openxla/stablehlo/blob/main/docs/reference_checklist.md#after-implementing-the-op.

### CODE REVIEW

File: stablehlo/dialect/StablehloOps.td [hunk 1]
```diff
@@ -1356,11 +1356,13 @@ def StableHLO_AllGatherOp : StableHLO_Op<"all_gather",
 
     Example:
     ```mlir
-    %result = "stablehlo.all_gather"(%operand) {
+    %result:2 = "stablehlo.all_gather"(%operand0, %operand1) {
       all_gather_dim = 1 : i64,
       replica_groups = dense<[[0, 1]]> : tensor<1x2xi64>,
+      // channel_id = 0
```
Comment (ghpvnist) [inline stablehlo/dialect/StablehloOps.td:1362]: Let's remove the default value (`channel_id` and `use_global_device_ids` from the td file).

**Review by ghpvnist — APPROVED** [review]
LGTM with one comment.

**Review by sdasgup3 — APPROVED** [review]

File: stablehlo/dialect/StablehloOps.td (hunk 1 above)
Comment (abhigunj) [reply stablehlo/dialect/StablehloOps.td:1362]: why ? 
from the spec too?

Comment (ghpvnist) [reply stablehlo/dialect/StablehloOps.td:1362]: Just some old decision made in 7.iii https://github.com/openxla/stablehlo/blob/main/docs/spec_checklist.md. FYI `use_global_device_ids` is a `UnitAttr` in MLIR, so should be written as `use_global_device_ids` (without the `=false`).

Comment (abhigunj) [reply stablehlo/dialect/StablehloOps.td]: Done, thanks for the reference.

**Pull request merged**